.github/workflows/ci.yaml

```diff
@@ -16,48 +16,52 @@ env:
   GIT_LFS_SKIP_SMUDGE: 1
 
 jobs:
-  build-lint-common:
+  common:
     if: github.event_name == 'pull_request'
-    ## Run only if a PR is made for the main branch
-    runs-on: buildjet-32vcpu-ubuntu-2204
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          ref: ${{ github.event.pull_request.head.sha }}
-          persist-credentials: false
-      - uses: cachix/install-nix-action@v24
-        with:
-          github_access_token: ${{ secrets.GITHUB_TOKEN }}
-      - name: Fmt + Doclint
-        run: nix build --log-lines 100 .\#checks.x86_64-linux.fmt .\#checks.x86_64-linux.doclint 
-      - name: Runtime common
-        run: nix build --log-lines 100 .\#checks.x86_64-linux.clippy-runtime-common .\#checks.x86_64-linux.nextest-runtime-common
-  build-lint-sydney:
-    if: github.event_name == 'pull_request'
-    ## Run only if a PR is made for the main branch
     runs-on: buildjet-32vcpu-ubuntu-2204
+    container: ggxdocker/ggxnode-dev:v1
     steps:
       - uses: actions/checkout@v4
         with:
           ref: ${{ github.event.pull_request.head.sha }}
           persist-credentials: false
-      - uses: cachix/install-nix-action@v24
-        with:
-          github_access_token: ${{ secrets.GITHUB_TOKEN }}
-      - run: nix build --log-lines 100 .\#checks.x86_64-linux.clippy-node-sydney .\#checks.x86_64-linux.nextest-sydney .\#checks.x86_64-linux.clippy-wasm-sydney
-  build-lint-brooklyn:
+
+      - uses: dsherret/rust-toolchain-file@v1
+
+      - name: fmt
```

Comment:
I would extract fmt to separate job. So the user would have more realistic check-up (including lint)

Reply:
That would require +1 executor (which may drive costs up)...